lib/ansible/modules/cloud/vmware/vmware_cluster_facts.py

```diff
@@ -218,6 +225,11 @@ def gather_cluster_facts(self):
             ha_vm_failure_interval = None
             enabled_vsan = False
             vsan_auto_claim_storage = False
+            hosts = []
+
+            # Hosts
+            for host in cluster.host:
+                hosts.append(host.name)
```

Comment:
I feel like it would be nice to also store the folder of the host. What about using a dict, and store in the name and the folder of the host? This way, we will be able to extend the information later with new fields, without changing the data structure.

Reply:
Do you mean this structure:
```yaml
hosts:
  esxi01.vsphere.local:
    folder: "/a/folder"
  esxi02.vsphere.local:
    folder: "/another/folder"
```

I am not sure about this as we do have a facts module for hosts. We need to make sure this module is only serving cluster facts. It may happen that this module gets too powerful and therefore slow.

Reply:
This would work, but I think the following structure sounds more natural.

```yaml
hosts:
    - name: esxi01.vsphere.local
      folder: '/a/folder'
    - name: esxi02.vsphere.local
      folder: '/another/folder'
```